File: comments/src/main/java/net/particify/arsnova/comments/model/serialization/UuidHelper.java
package net.particify.arsnova.comments.model.serialization;

import java.util.UUID;
import java.util.regex.Pattern;

public class UuidHelper {
  private static final String REPLACEMENT_PATTERN = "$1-$2-$3-$4-$5";
  private static final Pattern pattern = Pattern.compile("(\\w{8})(\\w{4})(\\w{4})(\\w{4})(\\w{12})");

  public static UUID stringToUuid(final String uuidString) {
    if (uuidString == null || uuidString.isEmpty()) {
      return null;
    }
    return UUID.fromString(pattern.matcher(uuidString).replaceFirst(REPLACEMENT_PATTERN));
  }

  public static String uuidToString(final UUID uuid) {
    return uuid == null ? null : uuid.toString().replace("-", "");
  }
}

File: authz/src/main/kotlin/net/particify/arsnova/authz/model/serialization/UuidHelper.kt
package net.particify.arsnova.authz.model.serialization

import java.util.UUID
import java.util.regex.Pattern

object UuidHelper {
  private const val REPLACEMENT_PATTERN = "$1-$2-$3-$4-$5"
  private val pattern = Pattern.compile("(\\w{8})(\\w{4})(\\w{4})(\\w{4})(\\w{12})")

  fun stringToUuid(uuidString: String?): UUID? {
    return if (uuidString.isNullOrEmpty()) {
      null
    } else {
      UUID.fromString(
        pattern.matcher(
          uuidString
        ).replaceFirst(REPLACEMENT_PATTERN)
      )
    }
  }

  @JvmStatic
  fun uuidToString(uuid: UUID?): String? {
    return uuid?.toString()?.replace("-", "")
  }
}
